source/common/buffer/buffer_impl.cc

```diff
@@ -115,7 +152,32 @@ ssize_t OwnedImpl::search(const void* data, uint64_t size, size_t start) const {
   return result_ptr.pos;
 }
 
-int OwnedImpl::write(int fd) { return evbuffer_write(buffer_.get(), fd); }
+int OwnedImpl::write(int fd) {
+  constexpr uint64_t MaxSlices = 16;
+  RawSlice slices[MaxSlices];
+  const uint64_t num_slices = std::min(getRawSlices(slices, MaxSlices), MaxSlices);
+  if (num_slices == 0) {
+    return 0;
```

Comment:
I think this return isn't needed. We'll return at ` if (num_slices_to_write == 0)` below